pintc/tests/storage/storage_map_keys.pnt

```diff
@@ -0,0 +1,46 @@
+union MyUnion = x | a(bool) | b(b256) | c(int) | d({ int, b256 }) | e({ b256, bool });
+type MyAlias = MyUnion;
+
+storage {
+    a: (int => b256),
+    b: (bool => int),
+    c: (b256 => bool),
+    d: ({ bool, int } => b256),
+    e: ({ int, b256 } => bool),
+    f: (MyUnion => int),
+    g: (int[5] => int),
+    h: ({ bool, b256 }[24][5] => int),
+    i: (MyAlias => bool),
+}
```

Comment:
Good.. now let's test these in `validation_tests` and `abi-gen-tests`.............
Sorry.........

Reply:
just added the `validation_tests`. On to `abi-gen-tests`